integration-tests/bats/status.bats

```diff
@@ -204,4 +204,95 @@ SQL
 @test "status: dolt reset --hard with more than one additional arg throws an error " {
     run dolt reset --hard HEAD HEAD2
     [ "$status" -eq 1 ]
+}
+
+@test "status: dolt reset hard with ~ works" {
```

Comment:
I think you should test other refspecs. Like a branch name, a commit hash, in addition to just ~

Reply:
Good point. Knew I was forgetting something